pkg/datapath/linux/node.go

```diff
@@ -1295,16 +1283,28 @@ func (n *linuxNodeHandler) NodeConfigurationChanged(newConfig datapath.LocalNode
 	n.nodeConfig = newConfig
 
 	if n.nodeConfig.EnableIPv4 {
+		ifaceName := ""
 		switch {
 		case option.Config.EnableNodePort:
 			mac, err := link.GetHardwareAddr(option.Config.DirectRoutingDevice)
 			if err != nil {
 				return err
 			}
+			ifaceName = option.Config.DirectRoutingDevice
 			n.enableNeighDiscovery = mac != nil // No need to arping for L2-less devices
 		case n.nodeConfig.EnableIPSec && option.Config.Tunnel == option.TunnelDisabled:
+			ifaceName = option.Config.EncryptInterface
 			n.enableNeighDiscovery = true
 		}
+
+		if n.enableNeighDiscovery {
+			link, err := netlink.LinkByName(ifaceName)
+			if err != nil {
+				return fmt.Errorf("cannot find link by name %s for neigh discovery: %w",
+					ifaceName, err)
+			}
+			n.neighDiscoveryLink = link
```

Comment:
Couldn't we do the following given that's all we ever use?
```suggestion
			n.neighDiscoveryLink = link.Attrs()
```

Reply:
Unfortunately we cannot, as `arping.PingOverLink()` expects `netlink.Link` instead of `netlink.LinkAttrs`.